### DISCUSSION TITLE
Probabilistic Programming

### DISCUSSION THREAD

**Discussion by saulshanabrook** [Quansight-Labs/metadsl#130] [Ideas]
Similar API to pymc4 (https://sharanry.github.io/post/eight-schools-model/) but not tied to tensorflow (can run on TF, numpy, pytorch)

**Comment by majidaldo** [comment]
I've been doing some research here. You could just wait until pymc4 gets a prototype developed (I sense there is some momentum to get something done and you would rely on them because they are experts).  Then see where the 'interfaces' are: model description (math description) --> model inference (computation). pymc3 was pretty tied to theano. I feel the same will happen with pymc4 but with tensorflow because the probabilistic computational basis has been developed in tensorflow probablility and edward2.
